packages/react-instantsearch-hooks/src/components/__tests__/routing/dispose-start.test.tsx

```diff
@@ -0,0 +1,108 @@
+import { render, waitFor } from '@testing-library/react';
+import historyRouter from 'instantsearch.js/es/lib/routers/history';
+import React, { useEffect } from 'react';
+import {
+  InstantSearch,
+  SearchBox,
+  useInstantSearch,
+} from 'react-instantsearch-hooks-web';
+
+import { createSearchClient } from '../../../../../../test/mock';
+
+describe('routing back and forth to an InstantSearch instance', () => {
+  test('updates the URL after the instance is disposed then restarted', async () => {
+    // -- Flow
+    // 1. Initial: '/'
+    // 2. Refine: '/?indexName[query]=Apple'
+    // 3. Dispose: '/'
+    // 4. Refine: '/'
+    // 5. Start: '/'
+    // 6. Refine: '/?indexName[query]=Apple'
+
+    const pushState = jest.spyOn(window.history, 'pushState');
+    const searchClient = createSearchClient({});
+
+    let setQuery = (_query: string) => {};
+    function QueryMiddleware() {
```

Comment:
wonder if this is clearer with a component using useSearchBox?

Reply:
See aa1644aa9627bfc770ec5cd0b2cd3c5eaf6cbec0.